lib/netsuite/async_result.rb

```diff
@@ -0,0 +1,59 @@
+# https://system.netsuite.com/help/helpcenter/en_US/Output/Help/SuiteCloudCustomizationScriptingWebServices/SuiteTalkWebServices/AsynchronousRequestProcessing.html
+module NetSuite
+  class AsyncResult
+
+    attr_reader :errors, :type
+
+    def initialize(async_result)
+      @type = async_result[:"@xsi:type"]
+
+      base_ref_list = []
+      errors = []
+
+      if async_result[:read_response_list]
+        response_list = async_result[:read_response_list][:read_response]
+      else
+        response_list = async_result[:write_response_list][:write_response]
+      end
+
+      Array[response_list].flatten.each do |response|
```

Comment:
:)  I like it.

On Wed, Feb 25, 2015 at 8:01 PM, Jerome Cornet <notifications@github.com>
wrote:

> In lib/netsuite/async_result.rb
> <https://github.com/prburke/netsuite/pull/2#discussion_r25396439>:
>
> > +
> > +    attr_reader :errors, :type
> > +
> > +    def initialize(async_result)
> > +      @type = async_result[:"@xsi:type"]
> > +
> > +      base_ref_list = []
> > +      errors = []
> > +
> > +      if async_result[:read_response_list]
> > +        response_list = async_result[:read_response_list][:read_response]
> > +      else
> > +        response_list = async_result[:write_response_list][:write_response]
> > +      end
> > +
> > +      Array[response_list].flatten.each do |response|
>
> very nitpicky, but a more ruby way to do this would be to do
>
> responses=Array[response_list].flatten.
> base_ref_list = responses.select { |response| response[:status][:@is_success] == 'true' }.map { |response| response[:base_ref] }
>
> errors = responses.reject { |response| response[:status][:@is_success] == 'true' }.map do |response|
>   details = Array[response[:status][:status_detail]].flatten.map { |details|  NetSuite::Error.new(details) }
>   [response[:base_ref][:@external_id], details]
> end
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prburke/netsuite/pull/2/files#r25396439>.
>